cmd/stellar-rpc/internal/ledgerentries/main.go

```diff
@@ -0,0 +1,91 @@
+package ledgerentries
+
+import (
+	"context"
+	"fmt"
+
+	coreProto "github.com/stellar/go/protocols/stellarcore"
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/stellar-rpc/cmd/stellar-rpc/internal/daemon/interfaces"
+	"github.com/stellar/stellar-rpc/cmd/stellar-rpc/internal/db"
+)
+
+type LedgerEntryGetter interface {
+	GetLedgerEntries(ctx context.Context, keys []xdr.LedgerKey) ([]db.LedgerKeyAndEntry, uint32, error)
+}
+
+// NewLedgerEntryGetter creates a LedgerEntryGetter which obtains the latest known value of the given ledger entries
+func NewLedgerEntryGetter(coreClient interfaces.FastCoreClient, latestLedgerReader db.LedgerEntryReader) LedgerEntryGetter {
+	return &coreLedgerEntryGetter{
+		coreClient: coreClient,
+		// use the latest ledger
+		atLedger:           0,
+		latestLedgerReader: latestLedgerReader,
+	}
+}
+
+// NewLedgerEntryAtGetter NewLedgerEntryGetter creates a LedgerEntryGetter which obtains the value of the given ledger entries at a fixed ledger
```

Comment:
nit: dupe in comment
```suggestion
// NewLedgerEntryGetter creates a LedgerEntryGetter which obtains the value of the given ledger entries at a fixed ledger
```